app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java

```diff
@@ -145,9 +145,9 @@ public void onError(Throwable e) {
                     public void onNext(Page<Client> clientPage) {
                         getMvpView().hideProgress();
                         if (clientPage.getPageItems().size() != 0) {
-                            long clientId = clientPage.getPageItems().get(0).getId();
-                            getMvpView().showClient(clientId);
-                            preferencesHelper.setClientId(clientId);
+                            Client clientCurrent = clientPage.getPageItems().get(0);
+                            getMvpView().showClient(clientCurrent);
```

Comment:
I do not think. we need this changes because for making request we need only client id that is enough for us. do not need to pass whole client.

Please revert back.